lib/cc/engine/analyzers/engine_config.rb

```diff
@@ -0,0 +1,49 @@
+module CC
+  module Engine
+    module Analyzers
+      class EngineConfig
+        attr_reader :config
+
+        def initialize(hash)
+          normalize(hash)
+        end
+
+        def language(language)
+          config.
+            fetch("languages", {}).
+            fetch(language, {})
+        end
+
+        def languages
+          config.fetch("languages", {})
+        end
+
+        private
+
+        attr_reader :hash
+
+        def normalize(hash)
+          @config = hash.fetch("config", {})
+          @config["languages"] = build_language_config(@config)
+        end
+
+        def build_language_config(config)
+          languages = config.fetch("languages", {})
+
+          if languages.is_a?(Array)
+            languages.each_with_object({}) do |language, map|
+              map[language.downcase] = {}
+            end
+          elsif languages.is_a?(Hash)
+            languages.each_with_object({}) do |language, map|
```

Comment:
What if we did something like `|(key, value), map|` here instead of doing the deconstructing in the line below?